### DISCUSSION TITLE
Installazioni multiple HA su stesso raspberry

### DISCUSSION THREAD

**Discussion by vinny74man** [indomus/forum#1523] [HUB: Home Assistant]
Ciao a tutti scusate se la domanda può sembrare un pò strana, ma ho un installazione molto vecchia di HA che fa il suo dovere con quello che attualmente c'è configurato, ma devo aggiungere dei dispositivi modbus e con il firmware attuale il modulo modbus non va, firmware 0.99, quindi vorrei provare ad eseguire una nuova installazione sempre sulla stessa macchina lasciando quella vecchia, per eseguire prima delle prove di funzionamento con i dispositivi più vecchi installati.

Pensate che sia possibile?
Attualmente HA è installato in venv con python 3.7, se questa nuova la install in docker cambiando la porta di ascolto possono andare in conflitto?

Grazie

**Comment by snakuzzo** [comment]
Guarda...lo puoi fare senza problemi. Crei un nuovo venv con python 3.9, crei un nuovo service sotto systemd e cambi la porta di ascolto di HA nel blocco `http:` del `configuration.yaml`.
Però permettimi di suggerirti di cambiare tipo di installazione. Passa a Docker...vedrai che sarà tutto molto più semplice e, anche volendo fare la prova che dici, ci metteresti un secondo a tirare su installazioni parallele.

**Comment by vinny74man** [comment]
Infatti stavo già vedendo come installare in docker e sicuramente questa è la strada che userò volevo solo essere sicuro di poter avere più istanze di HA senza che andassero in conflitto
Grazie

Scarica Outlook per Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by snakuzzo** [reply]
Per favore non rispondere dalla mail e fai reply sotto l'ultima risposta, non creando altro thread nella stessa discussione.
Comunque vai tranquillo...ne puoi creare quante ne vuoi

**Reply by vinny74man** [reply]
Scusa ho risposto dal cell e non ho fatto caso.
Installato e perfettamente funzionante, grazie per l'assistenza. 
Però se mi consenti voglio ancora approfittare della tua conoscenza, collegandomi in ssh al container c'è la possibilità di installare un editor di testo, io di solito uso vim o nano ma non sono presenti nessuno dei due.
Grazie

**Reply by snakuzzo** [reply]
Forse ti sfugge qualcosa e ti consiglio di studiare un attimo come funziona Docker, perché la tua domanda è sbagliata.
Non devi entrare nel container per editare i file... e non devi installare nulla all'interno del container.
I file li editi direttamente dall'host andando sulla directory che hai mappato come volume.
Ripeto...ti consiglio di studiare un attimo e di non andare a caso.

**Reply by vinny74man** [reply]
Si li modifico già in questo modo, ma ho visto che c'è la possibilità di collegarmi in ssh anche al container mi chiedevo se c'era questa possibilità ma se non è possibile non importa
Grazie

**Reply by snakuzzo** [reply]
Innanzi tutto devi sapere che di sicuro hai già `vi`, ma poi devi capire che quando distruggi un container per ricrearlo (tipico caso di aggiornamento di HA, o per una qualsiasi modifica che vuoi applicare al container), perdi quello che hai installato all'interno e dovresti reinstallarlo.
NON ha alcun senso logico fare quelo che vuoi fare.
Ti ripeto...devi studiare Docker e fare un po' di pratica, altrimenti certi concetti non ti saranno mai chiari al 100%

**Reply by snakuzzo** [reply]
P.S. e non è che ti colleghi in SSH al container. Esegui la shell del container... è una cosa diversa

**Reply by vinny74man** [reply]
Si c'è vi l'ho visto ma preferisco vim, però ho capito che non è indispensabile, di solito non uso docker, non ne ho mai avuto la necessità, però sembra che ormai sembra stia diventando lo standard per le installazioni di HA mi sa che devo iniziare a studiarmelo un pò.
Grazie

**Reply by snakuzzo** [reply]
Beh...diciamo che è la svolta un po' per qualsiasi tipo di installazione in qualsiasi ambito...anche e soprattutto a livello di sviluppo software

**Reply by vinny74man** [reply]
Io non sono uno sviluppatore, per me è un hobby che però a volte mi da anche molte soddisfazioni sul lavoro, sono un installatore, impianti antintrusione, TVCC e impianti domotoci, e con HA ho fatto risparmiare un pò di soldi ai miei clienti fornendo comunque un ottimo servizio, di solito eseguo impianti KNX e un web server di ABB per il kNX costa circa 1300€, con un raspi e un pò di volontà faccio tutto uguale e anche più bello.

**Reply by snakuzzo** [reply]
Sì chiaro, ma io non facevo un discorso da sviluppatore (cosa che non sono).  Se ne hai la possibilità perdici un po'di tempo per studiare soprattutto la parte dei volumi. Capita quella ti si aprirà un mondo :)

**Reply by vinny74man** [reply]
No figurati era solo per parlare, sicuramente lo studierò almeno qualcosina bisogna saperla se devi utilizzare un qualsiasi strumento o programma.
